src/cpp/session/SessionConsoleInput.cpp

```diff
@@ -156,11 +157,127 @@ void reissueLastConsolePrompt()
 
 void clearConsoleInputBuffer()
 {
-   // Discard any buffered input
-   while (!s_consoleInputBuffer.empty())
-      s_consoleInputBuffer.pop();
+   s_consoleInputBuffer = {};
+}
+
+namespace {
+
+// this function takes the next chunk of (potentially multi-line) pending
+// console input in the queue, then splits it into separate pieces of console
+// input with one piece of input for each line.
+//
+// we also fix up indentation if we can determine that the
+// code is going to be sent to the reticulate Python REPL.
+void fixupPendingConsoleInput()
+{
+   // get next input
+   auto input = s_consoleInputBuffer.front();
+   
+   // nothing to do if this is a cancel
+   if (input.cancel)
+      return;
+   
+   // if this has no newlines, then nothing to do
+   auto index = input.text.find('\n');
```

Comment:
Minor, could simplify to `if (input.text.find('\n') == std::string::npos)` and eliminate `index` variable.